src/main/scala/firrtl/Parser.scala

```diff
@@ -44,20 +44,31 @@ case class InvalidEscapeCharException(message: String) extends ParserException(m
 
 object Parser extends LazyLogging
 {
+  private def time[R](name: String)(block: => R): R = {
+    logger.info(s"Starting $name")
+    val t0 = System.nanoTime()
+    val result = block
+    val t1 = System.nanoTime()
+    logger.info(s"Finished $name")
+    val timeMillis = (t1 - t0) / 1000000.0
+    logger.info(f"$name took $timeMillis%.1f ms\n")
+    result
+  }
+
   /** Takes Iterator over lines of FIRRTL, returns AST (root node is Circuit)
     *
     * Parser performs conversion to machine firrtl
     */
   def parse(filename: String, lines: Iterator[String], useInfo: Boolean = true): Circuit = {
-    val fixedInput = Translator.addBrackets(lines)
+    val fixedInput = time("Translator") { Translator.addBrackets(lines) }
     //logger.debug("Preprocessed Input:\n" + fixedInput.result) 
     val antlrStream = new ANTLRInputStream(fixedInput.result)
     val lexer = new FIRRTLLexer(antlrStream)
     val tokens = new CommonTokenStream(lexer)
     val parser = new FIRRTLParser(tokens)
 
     // FIXME Dangerous (TODO)
```

Comment:
I should remove this comment. It was due to my misreading the ANTLR4 documentation. Using PredictionMode.LL was too slow, so I switched it to SLL, but I thought SLL could return incorrect results. It turns out SLL _can_ error on correct inputs, but still will always error on incorrect inputs. Erroring on correct inputs is a property of the grammar, and since we have yet to run into this, I think we're probably okay.

See:  http://www.antlr.org/api/Java/org/antlr/v4/runtime/atn/PredictionMode.html#enum_constant_detail